mapillary.py

```diff
@@ -0,0 +1,138 @@
+import time
+from dateutil import parser
+import requests
+import os
+import pickle
+import multiprocessing
+import util
+
+SEQUENCES_PER_PAGE_DEFAULT = 50  # How many sequences to receive on each page of the API call
+SKIP_IF_FEWER_IMAGES_THAN_DEFAULT = 10  # We will skip any sequences if they have fewer than this number of images
+SEQUENCE_URL = 'https://a.mapillary.com/v3/sequences_without_images?client_id={}&bbox={}&per_page={}'
+IMAGES_URL = 'https://a.mapillary.com/v3/images?client_id={}&sequence_keys={}'
+
+
+def run(bbox: str, traces_source: any, processes: int):
+    # Do a quick check to see if user specified the mandatory 'client_id' in traces_source JSON
+    if 'client_id' not in traces_source:
+        raise KeyError('Missing "client_id" (Mapillary Client ID) key in --traces-source JSON.')
+
+    # Create dirs
+    output_dir, output_tmp_dir = util.initialize_dirs(bbox)
+
+    # Break the bbox into sections and save it to a pickle file
+    bbox_sections = util.split_bbox(output_dir, bbox, to_bbox)
+
+    finished_bbox_sections = multiprocessing.Value('i', 0)
+    with multiprocessing.Pool(initializer=initialize_multiprocess,
+                              initargs=(output_dir, output_tmp_dir, traces_source, finished_bbox_sections),
+                              processes=processes) as pool:
+        result = pool.map_async(pull_and_save_trace_for_bbox, bbox_sections)
+
+        print('Placing {} results in {}...'.format(len(bbox_sections), output_dir))
+        progress = 0
+        increment = 5
+        while not result.ready():
+            result.wait(timeout=5)
+            next_progress = int(finished_bbox_sections.value / len(bbox_sections) * 100)
+            if int(next_progress / increment) > progress:
+                print('Current progress: {}%'.format(next_progress))
+                progress = int(next_progress / increment)
+        if progress != 100 / increment:
+            print('Current progress: 100%')
```

Comment:
@kevinkreiser This is the new multiprocessing section, basically we run the API trace pulling logic in parallel on the list of `bbox_sections`. I took the ideas of the "progress" stuff from `run_with_server.py`.

Reply:
yep looks good to me!